src/arithmetic/curves.rs

```diff
@@ -87,10 +93,8 @@ pub trait CurveAffine:
     type ScalarExt: FieldExt;
     /// The base field over which this elliptic curve is constructed.
     type Base: FieldExt;
-
-    /// Personalization of BLAKE2b hasher used to generate the uniform
-    /// random string.
-    const BLAKE2B_PERSONALIZATION: &'static [u8; 16];
+    /// The projective form of the curve
+    type CurveExt: CurveExt<AffineExt = Self, ScalarExt = <Self as CurveAffine>::ScalarExt>;
 
     /// CURVE_ID used for hash-to-curve.
     const CURVE_ID: &'static str;
```

Comment:
I suggested at https://github.com/zcash/halo2/pull/202#discussion_r580572834 that it might be simpler to move `CURVE_ID` to `CurveExt`.